#uint64 timestamp
uint8 state
